### PR TITLE
added add item section

### PR DESCRIPTION
**Pull request by ishwork**

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tanbt) [inline .idea/vcs.xml:1]: Please update `.gitignore` file to skip the whole `.idea`.
Also remove this folder (and its files) from the PR.

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "react-redux": "^7.2.2",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.0",
+    "reactstrap": "^8.7.1",
```
Comment (tanbt) [inline package.json:17]: I saw only the `Button` component from this `reactstrap` is used.
Is there any special reason that you want to use `Button` in this library?

Why don't use `Button` component from material-ui? You can check how I used it in `Home.js` file.

File: src/components/ItemDescription.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                <ItemList
+                  add={this.addNewRow}
+                  delete={this.clickOnDelete.bind(this)}
+                  itemLists={itemLists}
+                />
+              </tbody>
+              <tfoot>
+                <tr>
+                  <td>
+                    <Button
```
Comment (tanbt) [inline src/components/ItemDescription.js:85]: Why the "plus" button appears twice?
image

File: src/components/ItemDescription.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import ItemList from "./ItemList";
+import { Paper } from "@material-ui/core";
+import { Button } from "reactstrap";
+import "./ItemDescription.css";
+export default class ItemDescription extends React.Component {
```
Comment (tanbt) [inline src/components/ItemDescription.js:6]: Using class and properties is an old way of defining React component.
Any reason why don't keep using the new way `export default function ...` as we have done for other components in this project?

**Review by tanbt — COMMENTED** [review]
Please check my comments.

The UI isn't responsive yet, that's ok for now, we'll fix responsive later. But can you just move the content into the center of the page, right now it's stay of the left in wide screens.
image

File: package.json (hunk 2 above)
Comment (ishwork) [reply package.json]: have removed reactstrap

File: src/components/ItemDescription.js (hunk 3 above)
Comment (ishwork) [reply src/components/ItemDescription.js]: used only one button now

File: src/components/ItemDescription.js (hunk 4 above)
Comment (ishwork) [reply src/components/ItemDescription.js:5]: don't have any special reason but got errors while using function components and could not fix them. working with class component is easier for me.

File: package.json (hunk 2 above)
Comment (ishwork) [reply package.json]: have used button from materila-ui and will be using from it in other tasks.

File: .idea/vcs.xml (hunk 1 above)
Comment (ishwork) [reply .idea/vcs.xml:1]: I have only updated 5 files, you can see the commit that i made in my branch. i merged my branch with the main branch but for some reason, could not send pull request to main branch.

Comment (tanbt) [reply .idea/vcs.xml:1]: You can't change the main branch by yourself.
Any change must be in a pull request (PR), the PR needs to be reviewed and approved by someone else, then the PR can be merged to the main branch.
This is a strict policy so we make sure that no one will make any mistake alone.

**Review by tanbt — COMMENTED** [review]
The `items` section is now in center, but some parts of the layout is broken.
This is also an example why we need a review before pushing changes directly to `main` branch.

That's why it is recommended to `const useStyles....` for styling, instead of a CSS file.
Some hints:
* use `const useStyle...`
* define a better name for CSS selectors so it won't conflict with others. Following BEM rules could be fine.
* we agreed to not create separate CSS files for each component but using a global css file in `index.css` file, please put all CSS into that single file.

image

File: src/components/ItemDescription.js (hunk 3 above)
Comment (tanbt) [reply src/components/ItemDescription.js]: Looks good!

**Pull request closed without merging**